universal-application-tool-0.0.1/app/services/export/CsvExporter.java

```diff
@@ -50,17 +54,22 @@ private void writeHeadersOnFirstExport(CSVPrinter printer) throws IOException {
    * the writer between calls. Since it is intended for many applications, this function is intended
    * to be called several times.
    */
-  public void export(Application application, Writer writer) throws IOException {
+  public void export(
+      Application application, ReadOnlyApplicantProgramService roApplicantService, Writer writer)
+      throws IOException {
     CSVPrinter printer = new CSVPrinter(writer, CSVFormat.DEFAULT.withFirstRecordAsHeader());
 
     this.writeHeadersOnFirstExport(printer);
 
+    ImmutableMap<Path, String> answerMap =
+        roApplicantService.getSummaryData().stream()
+            .flatMap(data -> data.scalarAnswersInDefaultLocale().entrySet().stream())
+            .collect(ImmutableMap.toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
     for (Column column : getColumns()) {
       switch (column.columnType()) {
         case APPLICANT:
-          Optional<String> value =
-              application.getApplicantData().readAsString(column.jsonPath().orElseThrow());
-          printer.print(value.orElse(EMPTY_VALUE));
+          String value = getValueFromAnswerMap(column, answerMap);
```

Comment:
Can you do the same thing for APPLICANT_OPAQUE, below?

Reply:
Sure thing.